### PR TITLE
The Paint & Linen Update [Splinter 6]: Flax, Linen, Weaving, Sewing, Dyeing

### PR DESCRIPTION
**Pull request by DeityLink**
286402277-8718cf05-f07b-42bb-8549-a583c734704b286404044-d750c1a8-8178-4416-b6bb-02cdffcfe628286409874-c088342b-ed8c-44d5-9e62-1ef497be8abe286410708-9e7c10c9-dc3b-4d3b-ad0a-c1995acc71ab286454032-22891c51-930e-499b-a84a-4be58b74ce1b

### CODE REVIEW

File: code/modules/reagents/reagent_dispenser.dm
Comment (Kurfursten) [inline code/modules/reagents/reagent_dispenser.dm]: much better to do this in an oop way like an overrideable proc or var, especially because bus happens

File: code/game/objects/items/stacks/stack_recipes.dm
Comment (Kurfursten) [inline code/game/objects/items/stacks/stack_recipes.dm:589]: since you are offering null as a possibility they can cancel here and there's no sanity against that (happens above too)

File: code/game/objects/structures/bedsheet_bin.dm
Comment (Kurfursten) [inline code/game/objects/structures/bedsheet_bin.dm]: maybe a feedback message to explain what you are doing here?

**Review by Kurfursten — COMMENTED** [review]
Reviewed 105 files today

File: code/modules/clothes_making/weaving.dm
Comment (Kurfursten) [inline code/modules/clothes_making/weaving.dm]: if 1, will say "1 more cloth sheets"; I think there's a macro to handle plurality agreement

File: code/modules/clothes_making/weaving.dm
Comment (Kurfursten) [inline code/modules/clothes_making/weaving.dm]: This whole sentence sounds wrong, try
"Grow flax and insert it to spin cloth."

File: code/modules/clothes_making/weaving.dm
Comment (Kurfursten) [inline code/modules/clothes_making/weaving.dm]: not a big deal but prefer these as SECONDS

File: code/modules/clothes_making/weaving.dm
Comment (Kurfursten) [inline code/modules/clothes_making/weaving.dm]: You mount the flax upon the loom.

File: code/modules/clothes_making/weaving.dm
Comment (Kurfursten) [inline code/modules/clothes_making/weaving.dm:84]: did you intend to play two overlapping sounds?

File: code/modules/clothes_making/weaving.dm
Comment (Kurfursten) [inline code/modules/clothes_making/weaving.dm:79]: seems like it might be nice to scale on robustness or some other attribute but this is fine

File: code/modules/clothes_making/weaving.dm
Comment (Kurfursten) [inline code/modules/clothes_making/weaving.dm:81]: might be fine but I have a vague memory that deleting something straight from a bag without calling remove_from_storage first can create ghost images if they had their bag open at the time - maybe we fixed this? please test

File: code/modules/clothes_making/weaving.dm
Comment (Kurfursten) [inline code/modules/clothes_making/weaving.dm]: As above, give this a grammar pass. Also:

`Flax can be loaded automatically by conveyor belt.`

File: code/modules/clothes_making/weaving.dm
Comment (Kurfursten) [inline code/modules/clothes_making/weaving.dm]: There are
eject them
it's
pick them

of course all of these imply plurality which it might not be, you could rephrase to work with any plurality with:
`There are [stored_cloth] lengths of cloth on the roll being spun. The machine will eject when it is full, or you can eject the roll now.`

File: code/modules/clothes_making/weaving.dm
Comment (Kurfursten) [inline code/modules/clothes_making/weaving.dm]: should probably return TRUE in case there is ever a loom subtype

File: code/modules/clothes_making/weaving.dm
Comment (Kurfursten) [inline code/modules/clothes_making/weaving.dm]: woven by

Comment (Kurfursten) [reply code/modules/clothes_making/weaving.dm]: this is repeated several times below as well

File: code/modules/clothes_making/weaving.dm
Comment (Kurfursten) [inline code/modules/clothes_making/weaving.dm]: this is not repeated inside the process, so if your methylin runs out you can continue weaving

also, user not usr

**Review by Kurfursten — CHANGES_REQUESTED** [review]
mostly typos

File: code/game/objects/items/stacks/stack_recipes.dm
Comment (Kurfursten) [inline code/game/objects/items/stacks/stack_recipes.dm:539]: carpet seems like an obvious thing to sew

Comment (DeityLink) [reply code/game/objects/items/stacks/stack_recipes.dm:539]: I'm definitely making a note of it, that's a good idea. But same as with the map changes, I'll wait until after the update gets merged as there's lots of testing and ideas I would need to explore, that would require the other features to be here as well. (namely if we can sew carpets those should be by default of the same color as the cloth roll, but you could dye them just like clothing, except it would basically update the tile's paint_overlay, but then I'd have to add a new proc to update the overlays' shape based on how the carpet turfs smooth with their neighbors)

File: code/game/objects/items/stacks/stack_recipes.dm
Comment (DeityLink) [reply code/game/objects/items/stacks/stack_recipes.dm:589]: yeah I forgot to have null taken into account for the second choice, it'll do the same as choosing "None", and you'll sew pants without top. If you cancel above it will just cancel the crafting altogether.

File: code/modules/clothes_making/weaving.dm
Comment (DeityLink) [reply code/modules/clothes_making/weaving.dm]: Not sure the macro handles it unless the word itself is a plural or something, but I'll add something to handle that case.

File: code/modules/clothes_making/weaving.dm
Comment (DeityLink) [reply code/modules/clothes_making/weaving.dm:84]: yes, in this specific case

File: code/modules/clothes_making/weaving.dm
Comment (DeityLink) [reply code/modules/clothes_making/weaving.dm:79]: it will scale on upgrades when using the electric loom. I intend the spinning loom to produce only as much as an un-upgraded machine will. (and upgraded machines can already produce a LOT of cloth).

Botanists may prefer to upgrade the plant's yield over its potency.

Higher potency however means more flax oil when ground.

File: code/modules/clothes_making/weaving.dm
Comment (DeityLink) [reply code/modules/clothes_making/weaving.dm:81]: Tested with the bag open, the flax got removed from the UI properly, guess it was fixed.

File: code/modules/clothes_making/weaving.dm
Comment (DeityLink) [reply code/modules/clothes_making/weaving.dm]: weaving on the electric loom isn't a continuous action unlike with the spinning wheel, attack_hand only serves to pick up the current cloth roll so a single dexterity check here is enough.

**Comment by DeityLink** [comment]
Applied the requested changes.

**Review by Kurfursten — APPROVED** [review]

**Comment by Kurfursten** [comment]
Github was chugging trying to open the code approval window for just 108 files so I can only imagine what would happen if you hadn't atomized this

**Comment by DeityLink** [comment]
hold on

**Comment by DeityLink** [comment]
there we go

**Pull request merged**
